### PR TITLE
get users subcommand

### PR DESCRIPTION
**Pull request by petedannemann**
This PR adds the `get users` subcommand. It displays the users in the cluster as well as some basic, non-sensitive credential information

Help command
```bash
$ topicctl get users --help
Displays information for all users in the cluster.

Usage:
  topicctl get users [flags]

Flags:
  -h, --help   help for users

Global Flags:
  -b, --broker-addr string       Broker address
      --cluster-config string    Cluster config
      --debug                    enable debug logging
      --expand-env               Expand environment in cluster config
      --full                     Show more full information for resources
      --no-spinner               disable all UI spinners
      --sasl-mechanism string    SASL mechanism if using SASL (choices: AWS-MSK-IAM, PLAIN, SCRAM-SHA-256, or SCRAM-SHA-512)
      --sasl-password string     SASL password if using SASL; will override value set in cluster config
      --sasl-username string     SASL username if using SASL; will override value set in cluster config
      --sort-values              Sort by value instead of name; only applies for lags at the moment
      --tls-ca-cert string       Path to client CA cert PEM file if using TLS
      --tls-cert string          Path to client cert PEM file if using TLS
      --tls-enabled              Use TLS for communication with brokers
      --tls-key string           Path to client private key PEM file if using TLS
      --tls-server-name string   Server name to use for TLS cert verification
      --tls-skip-verify          Skip hostname verification when using TLS
  -z, --zk-addr string           ZooKeeper address
      --zk-prefix string         Prefix for cluster-related nodes in zk
```
Usage:
```bash
$ topicctl get users --cluster-config examples/auth/cluster.yaml
[2023-09-19 15:54:49]  INFO Users:
-------------+-----------+-------------
     NAME    | MECHANISM | ITERATIONS
-------------+-----------+-------------
  adminscram | sha256    |       4096
  adminscram | sha512    |       4096
  junk       | sha512    |      15000
-------------+-----------+-------------
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,7 +11,7 @@ require (
 	github.com/hashicorp/go-multierror v1.1.1
 	github.com/olekukonko/tablewriter v0.0.5
 	github.com/samuel/go-zookeeper v0.0.0-20190923202752-2cc03de413da
-	github.com/segmentio/kafka-go v0.4.35
```
Comment (petedannemann) [inline go.mod:14]: Need to release a new version of kafka-go with the added ACLs APIs and bump the version here

File: docker-compose-auth.yml [hunk 2]
```diff
@@ -16,7 +16,7 @@ services:
       - "2181:2181"
 
   kafka:
-    image: wurstmeister/kafka:2.12-2.4.1
+    image: wurstmeister/kafka:2.13-2.7.1
```
Comment (petedannemann) [inline docker-compose-auth.yml:19]: Need to decide if we are ok with bumping the test version and also get this tested in CI

**Comment by petedannemann** [comment]
Need to add this to the repl

File: pkg/admin/brokerclient.go [hunk 3]
```diff
@@ -104,6 +104,12 @@ func NewBrokerAdminClient(
 	if _, ok := maxVersions["DescribeAcls"]; ok {
 		supportedFeatures.ACLs = true
 	}
+
+	// If we have DescribeUserScramCredentials, than we're running a version of Kafka > 2.7.1,
```
Comment (hhahn-tw) [inline pkg/admin/brokerclient.go:108]: than -> then

File: pkg/admin/brokerclient.go [hunk 4]
```diff
@@ -379,6 +385,72 @@ func (c *BrokerAdminClient) GetTopic(
...
+
+func (c *BrokerAdminClient) CreateUser(
+	ctx context.Context,
+	user kafka.UserScramCredentialsUpsertion,
+) error {
+	if c.config.ReadOnly {
+		return errors.New("Cannot create user in read-only mode")
+	}
+	req := kafka.AlterUserScramCredentialsRequest{
+		Upsertions: []kafka.UserScramCredentialsUpsertion{user},
```
Comment (hhahn-tw) [inline pkg/admin/brokerclient.go:440]: This gives me the impression updates and inserts are a single function for kafka. If so, should this function be named something other than CreateUser? It seems like it'd handle updates as-is, for example.

Comment (petedannemann) [reply pkg/admin/brokerclient.go:440]: Good call, I renamed it to `UpsertUser`

**Review by hhahn-tw — APPROVED** [review]

**Pull request merged**